File: 3278-find-the-number-of-ways-to-place-people-i/find-the-number-of-ways-to-place-people-i.py
from typing import List
# for type hints such as List[List[int]] -> int

class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        points.sort(key = lambda point: (point[0], -point[1]))

        n = len(points)
        result = 0
        for i in range(0, n, 1):
            y = float("-inf")
            for j in range(i + 1, n, 1):
                if points[i][1] >= points[j][1] and points[j][1] > y:
                    result += 1
                    y = points[j][1]
        
        return result

File: 3278-find-the-number-of-ways-to-place-people-i/find-the-number-of-ways-to-place-people-i.cpp
class Solution {
public:
    int numberOfPairs(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b) {
            if (a[0] != b[0]) {
                return a[0] < b[0];
            } else {
                return a[1] > b[1];
            }
        });

        int n = points.size(), result = 0;

        for (int i = 0; i < n; i++) {
            int y = INT_MIN;
            for (int j = i + 1; j < n; j++) {
                    if (points[i][1] >= points[j][1] && points[j][1] > y) {
                    result += 1;
                    y = points[j][1];
                }
            }
        }

        return result;
    }
};